sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuShuffleExchangeExec.scala

```diff
@@ -62,6 +62,9 @@ class GpuShuffleMeta(
           .foreach { dataType =>
             willNotWorkOnGpu(s"round-robin partitioning cannot sort $dataType")
           }
+    } else if (shuffle.outputPartitioning.isInstanceOf[HashPartitioning] &&
+        shuffle.output.exists(_.dataType.isInstanceOf[ArrayType])) {
```

Comment:
I'm not sure this fixes all cases.  What about a struct of array types? The reason the RoundRobin one could be simple because both structs and arrays were not allowed so all nested types were eliminated. It might be simpler to use TypeSig to show what is allowed.

```
val allowed = (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.STRUCT).nested()
shuffle.output.foreach { att => 
  if (!allowed.isSupportedType(att.dataType)) {
    willNotWorkOnGpu("hash partitioning does not support ${att.dataType}")
  }
}
```

Reply:
Also in both your check and my check we are really being overly restrictive.  The only cases where this is an issue is an Array with an Array or string under it.  So if we really wanted to allow as much as possible we could write something specific that walks through all of the nested types looking for arrays, and if it finds one, then it switches to look under it for arrays or strings, and if it finds those, then it fails.

Reply:
I've refined the check to only exclude arrays of nested types and strings.